src/machine.rs

```diff
@@ -140,6 +140,7 @@ impl Memory {
     }
 }
 
+// The VM which stores the state of interpreting the MIR
```

Comment:
This is pretty hard to parse (and should it be `///`?). Can we try rephrasing?

Reply:
I've given it another go, it's a little hard to come up with a comment for this without saying anything that sounds 'too obvious'.